source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -105,6 +158,73 @@ Extractor::extractValue(Http::StreamFilterCallbacks &callbacks,
   return "";
 }
 
+// Match a regex against the input value and replace the matched subgroup with the replacement_text_ value
+std::string
+Extractor::replaceIndividualValue(Http::StreamFilterCallbacks &callbacks,
+                                  absl::string_view value) const {
+  std::match_results<absl::string_view::const_iterator> regex_result;
+
+  // if there are no matches, return an empty string
+  if (!std::regex_search(value.begin(), value.end(), regex_result, extract_regex_)) {
+    ENVOY_STREAM_LOG(debug, "replaceValue: extractor regex did not match input", callbacks);
+    return "";
+  }
+
+  // if the subgroup specified is greater than the number of subgroups in the regex, return an empty string
+  if (group_ >= regex_result.size()) {
+    // this should never happen as we test this in the ctor.
+    ASSERT("no such group in the regex");
+    ENVOY_STREAM_LOG(debug, "replaceValue: invalid group specified for regex", callbacks);
+    return "";
+  }
+
+  // if the regex doesn't match the entire input value, return an empty string
+  if (regex_result[0].length() != long(value.length())) {
+    ENVOY_STREAM_LOG(debug, "replaceValue: Regex did not match entire input value. This is not allowed in SINGLE_REPLACE mode.", callbacks);
+    return "";
+  }
+
+  // Create a new string with the maximum possible length after replacement
+  auto max_possible_length = value.length() + replacement_text_.value().length();
+  std::string replaced;
+  replaced.reserve(max_possible_length);
+
+  auto subgroup_start = regex_result[group_].first;
+  auto subgroup_end = regex_result[group_].second;
+
+  // Copy the initial part of the string until the match
+  replaced.assign(value.begin(), subgroup_start);
+
+  // Append the replacement text
+  replaced += replacement_text_.value();
+
+  // Append the remaining part of the string after the match
+  replaced.append(subgroup_end, value.end());
+
+  return replaced;
+}
+
+// Match a regex against the input value and replace all instances of the regex with the replacement_text_ value
+std::string
+Extractor::replaceAllValues(Http::StreamFilterCallbacks &callbacks,
+                            absl::string_view value) const {
+  std::string input(value.begin(), value.end());
+  std::string replaced;
+
+  // create an iterator to search for matches of extract_regex_ in the input string
+  std::sregex_iterator it(input.begin(), input.end(), extract_regex_);
+  std::sregex_iterator end_it; // default end iterator for comparison
+
+  // return an empty string if the regex doesn't match any part of the input value
+  if (it == end_it) {
+    ENVOY_STREAM_LOG(debug, "replaceAllValues: extractor regex did not match input", callbacks);
+    return "";
+  }
+
```

Comment:
I see that this function returns the whole input, and not just the parts matched by the regex. 
If that's the case, we can keep this semantic even when the regex doesn't match, and remove this block?

Reply:
Removing this block causes this mode (REPLACE_ALL) to behave differently than the other modes in the case where the regex does not match the input

With EXTRACT (i.e. previous default extractor behavior) and SINGLE_REPLACE modes, if the regex does not match the input, we return an empty string. With the suggested change, if the regex does not match the input, we return the input.

Basically, I'm concerned that it could be confusing for users who are debugging problematic extractors to have behaviors vary between different modes.

I'm still open to revising and/or removing this block, but that's why it's in place

Reply:
Updated this here: 6469b2614ffaaa98a82ccce921bdad77e89aacd9

Now, when the regex doesn't match in `REPLACE_ALL` mode, we return the input instead of an empty string